fuzz/fuzz_targets/parsing_context_parse_and_resolve.rs

```diff
@@ -0,0 +1,24 @@
+#![no_main]
+
+use std::io::Write;
+
+use libfuzzer_sys::fuzz_target;
+
+use solar_interface::Session;
+use tempfile::NamedTempFile;
+
+fuzz_target!(|data: &[u8]| {
+    let mut file = NamedTempFile::new().expect("Failed to create named temporary file");
+    file.write_all(data).expect("Failed to write to temporary file");
+
+    let path = file.into_temp_path();
+    let path = path.keep().expect("Failed to persist temporary file");
+
+    let sess = Session::builder().with_buffer_emitter(solar_interface::ColorChoice::Auto).build();
+
+    let _ = sess.enter(|| -> solar_interface::Result<()> {
+        let mut pcx = solar_sema::ParsingContext::new(&sess);
+        pcx.load_file(&path).unwrap();
```

Comment:
We can avoid the temp file by using `add_file` with a `SourceFile` created in memory, like so:
```rust
let Ok(data) = std::str::::from_utf8(data) else {
    return Ok(());
};
let Ok(file) = sess.source_map().new_dummy_source_file("fuzz".into(), data.to_string()) else {
    return;
};
pcx.add_file(file);
```